lib/fastq_utils.py

```diff
@@ -16,6 +16,10 @@
 from pathlib import Path
 import requests
 
+hostile_pypath = list(Path(os.environ['KB_RUNTIME']).glob('hostile/lib/python3.*/site-packages'))[-1]
+sys.path.append(str(hostile_pypath))
```

Comment:
Why is this needed in the path? I don't see any executables there related to hostile:

```
>>> list(Path(os.environ['KB_RUNTIME']).glob('hostile/lib/python3.*/site-packages'))[-1]
PosixPath('/opt/patric-common/runtime/hostile/lib/python3.10/site-packages')

Singularity> ls -l /opt/patric-common/runtime/hostile/lib/python3.10/site-packages | grep x|grep -v '^d'
-rwxrwxr-x 2 root root 215248 Sep 29  2023 _cffi_backend.cpython-310-x86_64-linux-gnu.so
-rw-rw-r-- 2 root root   6227 Feb  4  2023 mypy_extensions.py
-rw-rw-r-- 3 root root    119 Mar 20 08:01 README.txt
-rw-rw-r-- 4 root root  34549 May  5  2021 six.py
-rw-rw-r-- 2 root root    574 Jul 31  2019 sphinxcontrib_napoleon-0.7-py2.7-nspkg.pth
-rw-rw-r-- 3 root root 122293 Apr  5 10:14 typing_extensions.py
```

Reply:
This enables python3 to access this Python library, via the Python API in the lines directly below that.
I thought I'd try it this way, rather than maybe add extra complication and bug surface area by going from Python, to subprocess, back to Python.
But let me double check that actually uses the right Python 3 version . . .

Reply:
ah I was thinking sys.path was executable path; however, if the package was installed into site packages it should be loadable without any manual manipulation of the path like this.

Reply:
Ah, now I see it. This wasn't installed in the primary python. Since hostile is using a venv, it's probably reasonable to think about using the venv for the execution of the python code. We could easily run into trouble trying to run modules installed in one python environment from another python environment.

Reply:
Either that or make a separate standalone script to do the hostile processing and we can ensure that is executed with the python that is set up inside the venv

Reply:
I just pushed an update which will call the "right" Python in the conda env via subprocess.
We already have it there. Might as well use it, I guess.

That's probably the safest, as that is consistent with the upstream pubished to conda-forge + bioconda, via their recommended install instructions.
https://github.com/bede/hostile/blob/cb98c9d8f21dd0fab4365e0d28b93ebe089de69e/README.md?plain=1#L31-L36
Although, it looks like he mainly does that to make sure to also have tools we already have (minimap2, etc., etc.)

It would be nice, and might be safe enough, to not have to spend so much container space on conda envs.
But with the Python community often not being very careful with semver, one often get's bitten by using "research grade" software, for "production".

Idk . . . We can try a venv . . .

Python is so painful, sometimes, in terms of versions (I suppose the same could be said of other languages, if they were as active among newbies.)
A Julia Project.toml file fully specifies the top level deps via uuid (and thus the entire dep tree). It is automatically generated when adding software. That can safely share assets (due to uuid).

But I digress.

Reply:
Yes. The entire ecosystem involved in having to spin up custom venvs or conda builds for each indivdual tool because nobody can do reliable release engineering on a per-tool basis or get forward or backward compatibility right is a huge mess for anyone trying to use more than one tool at a time. At this point I do not care at all about the space taken up by the multiple conda builds. It is more important that the system be stable.